openstack/dws/cluster/requests.go

```diff
@@ -57,6 +84,60 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
 			"keep_last_manual_snapshot": 0,
 		},
 	}
+
 	_, r.Err = c.Delete(resourceURL(c, id), reqOpt)
 	return
 }
+
+func List(c *golangsdk.ServiceClient, opts ListOpts) (*ListClustersRst, error) {
+	var rst ListClustersRst
+	q, err := golangsdk.BuildQueryString(opts)
+	if err != nil {
+		return nil, err
+	}
+	url := listURL(c) + q.String()
+
+	_, err = c.Get(url, &rst, &golangsdk.RequestOpts{MoreHeaders: RequestOpts.MoreHeaders})
+	if err == nil {
+		return &rst, nil
+	}
+
+	return nil, err
+}
+
+func ResetPassword(c *golangsdk.ServiceClient, clusterId string, opts ResetPasswordOpts) (*golangsdk.Result, error) {
+	b, err := golangsdk.BuildRequestBody(opts, "")
+	if err != nil {
+		return nil, err
+	}
+
+	var r golangsdk.Result
+	_, err = c.Post(resetPasswordURL(c, clusterId), b, &r.Body, &golangsdk.RequestOpts{
+		MoreHeaders: RequestOpts.MoreHeaders,
+	})
+	return &r, err
+}
+
+func Resize(c *golangsdk.ServiceClient, clusterId string, scaleOutCount int) (*golangsdk.Result, error) {
+	var requestBody = map[string]interface{}{
+		"scale_out": map[string]interface{}{
+			"count": scaleOutCount,
+		},
+	}
+
+	var r golangsdk.Result
+	_, err := c.Post(resizeURL(c, clusterId), requestBody, &r.Body, &golangsdk.RequestOpts{
+		MoreHeaders: RequestOpts.MoreHeaders})
+	return &r, err
+}
+
+func ListNodeTypes(c *golangsdk.ServiceClient) (*NodeTypes, error) {
+	var rst NodeTypes
+	_, err := c.Get(listNodeTypesURL(c), &rst, &golangsdk.RequestOpts{
+		MoreHeaders: RequestOpts.MoreHeaders,
+	})
+	if err == nil {
+		return &rst, nil
+	}
+	return nil, err
```

Comment:
ditto

Reply:
if exist err, expect to return null rather than an empty obj